lib/src/generate/build_impl.dart

```diff
@@ -79,27 +83,45 @@ class BuildImpl {
     }
   }
 
-  /// Asset containing previous build outputs.
-  AssetId get _buildOutputsId =>
-      new AssetId(_packageGraph.root.name, '.build/build_outputs.json');
+  /// Asset containing previous asset dependency graph.
+  AssetId get _assetGraphId =>
+      new AssetId(_packageGraph.root.name, '.build/asset_graph.json');
+
+  /// Reads in the [assetGraph] from disk.
+  Future _readAssetGraph() async {
+    if (!await _reader.hasInput(_assetGraphId)) return new AssetGraph();
+    try {
+      _logger.info('Reading cached asset graph.');
+      var graph = new AssetGraph.deserialize(
+          JSON.decode(await _reader.readAsString(_assetGraphId)));
+      /// TODO(jakemac): Only invalidate nodes which need invalidating, which
+      /// will give us incremental builds on startup.
+      var generatedNodes =
+          graph.allNodes.where((node) => node is GeneratedAssetNode);
+      for (var node in generatedNodes) {
+        node.needsUpdate = true;
+      }
+      return graph;
+    } on AssetGraphVersionException catch (_) {
+      return new AssetGraph();
```

Comment:
In context, this is a reasonable return, but taken alone it has a bad code smell. Add a comment explaining why ignoring the error & returning an empty `AssetGraph` is the right way to handle this.

Is it worthwhile to log the error?